plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/join/JoinProcessor.java

```diff
@@ -68,8 +72,14 @@ public String getType() {
     public static class Factory implements Processor.Factory<JoinProcessor> {
         @Override
         public JoinProcessor create(Map<String, Object> config) throws Exception {
-            Map<String, String> fields = ConfigurationUtils.readMap(config, "fields");
-            return new JoinProcessor(Collections.unmodifiableMap(fields));
+            if (config.isEmpty()) {
+                throw new IllegalArgumentException("processor [" + JoinProcessor.TYPE + "] requires a field to be specified.");
+            } else if (config.size() != 1) {
+                throw new IllegalArgumentException("processor [" + JoinProcessor.TYPE + "] only accepts one field. Too many fields were provided: " +  Arrays.toString(config.keySet().toArray()));
+            }
+            String field = (String) config.keySet().toArray()[0];
```

Comment:
> toArray() returns an Object[]

depends on which toArray method you use. Just move to the one that accept an array as argument. Or iterator is fine too.